internal/data/circle_transfer_requests.go

```diff
@@ -175,6 +175,48 @@ func (m CircleTransferRequestModel) Get(ctx context.Context, sqlExec db.SQLExecu
 	return &circleTransferRequests, nil
 }
 
+func (m CircleTransferRequestModel) GetCurrentTransfersForPaymentIDs(ctx context.Context, sqlExec db.SQLExecuter, paymentIDs []string) (map[string]*CircleTransferRequest, error) {
+	if len(paymentIDs) == 0 {
+		return nil, fmt.Errorf("paymentIDs is required")
+	}
+
+	query := `
+		SELECT DISTINCT ON (payment_id)
+			idempotency_key,
+			payment_id,
+			circle_transfer_id,
+			status,
+			response_body,
+			source_wallet_id,
+			created_at,
+			updated_at,
+			completed_at
```

Comment:
Can you select all calumns with `*`?
```suggestion
			*
```

Reply:
addressed in 566eee114138c34288c45f88050d1626980213d2